sw/device/lib/dif/dif_aes.c

```diff
@@ -267,11 +283,69 @@ dif_result_t dif_aes_read_output(const dif_aes_t *aes, dif_aes_data_t *data) {
     return kDifError;
   }
 
-  for (int i = 0; i < AES_DATA_OUT_MULTIREG_COUNT; ++i) {
-    ptrdiff_t offset = AES_DATA_OUT_0_REG_OFFSET + (i * sizeof(uint32_t));
+  aes_read_multireg(aes, data->data, AES_DATA_OUT_MULTIREG_COUNT,
+                   AES_DATA_OUT_0_REG_OFFSET);
 
-    data->data[i] = mmio_region_read32(aes->base_addr, offset);
+  return kDifOk;
+}
+
+dif_result_t dif_aes_process_data(const dif_aes_t *aes,
+                                  const dif_aes_data_t *plain_text,
+                                  dif_aes_data_t *cipher_text,
+                                  size_t block_amount) {
```

Comment:
`amount` of what? This doesn't make it clear what amount related to a block (block size, block count) we're talking about here, so I'd go with `block_count` to be more consistent with the rest of the project.